src/screens/CheckIdentifierScreen.tsx

```diff
@@ -0,0 +1,67 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useAndroidBackHandler } from 'react-navigation-backhandler';
+
+import styled from '@/styled-thing';
+import { fonts, position } from '@/styles';
+import { ThemeContextProps } from '@/theme';
+import { SheetActionButton, SheetTitle } from '@/components/sheet';
+import { isAuthenticated } from '@/utils/authentication';
+import { RouteProp, useRoute } from '@react-navigation/native';
+import { RootStackParamList } from '@/navigation/types';
+import { checkKeychainIntegrity } from '@/redux/wallets';
+import store from '@/redux/store';
+
+// @ts-expect-error Our implementation of SC complains
+const Container = styled.View({
+  ...position.coverAsObject,
+  alignItems: 'center',
+  backgroundColor: ({ theme: { colors } }: { theme: ThemeContextProps }) => colors.alpha(colors.white, 0.9),
+  justifyContent: 'center',
+});
+
+export default function CheckIdentifierScreen() {
+  const { params } = useRoute<RouteProp<RootStackParamList, 'CheckIdentifierScreen'>>();
+
+  const { onSuccess, onFailure } = params;
+
+  const [isChecking, setIsChecking] = useState(false);
+
+  useAndroidBackHandler(() => {
+    return true;
+  });
+
+  const initAndRunKeychainChecks = useCallback(async () => {
+    if (isChecking) return;
+    setIsChecking(true);
+
+    try {
+      const authenticated = await isAuthenticated();
+      console.log({ authenticated });
+      if (!authenticated) return;
+
+      // TODO: what causes an unhealthy wallet?
+
+      // TODO: Do some checklist animations here while we're checking integrity?
+      const isHealthy = await store.dispatch(checkKeychainIntegrity());
+      if (isHealthy) return onSuccess();
```

Comment:
I know we said checking keychain integrity and we have a function like that but it's much more complex than what we need and has a different intended usage.

I'd do this instead:

read  `rainbowAddressKey` from the public keychain. Let's call it `selectedAddress`
If that exists I'd try to read `"${selectedAddress}_rainbowPrivateKey"` from the PRIVATE keychain. This will require auth automatically.

If that exists and u get a valid pkey, then we know it's an reinstall (and not a phone migration) because the private keychain is still available.